tensorflow/compiler/xla/service/gpu/ir_emitter_unnested.cc

```diff
@@ -3056,6 +3066,14 @@ ReductionCodegenInfo IrEmitterUnnested::ComputeReductionCodegenInfo(
 
   int64 num_threads_y = reduction_dimensions.is_row_reduction ? 1 : kWarpSize;
   int64 num_threads_x = kWarpSize;
+  if (reduction_dimensions.is_row_reduction) {
+    int cc_major = 0, cc_minor = 0;
+    ir_emitter_context_->device_description().cuda_compute_capability(
+        &cc_major, &cc_minor);
+    if (cc_major >= 6 && smallest_input_dtype_bits <= 16) {
+      num_threads_x = kWarpSize * 4;
```

Comment:
Sorry for colliding here: in the meantime I have committed https://github.com/tensorflow/tensorflow/commit/0c3feb1cadf79420b18cc0ce7731476ba45d00df which started using intra-block communication for row reduction, could you check whether these numbers need updating?

Reply:
I rebased, fixed the review and I pushed the new code. Running all test cases is long. I'll probably have V100 results tomorrow. But for P100, it will probably take a few days.

I have another PR that I should make this week. Do you plan to change this code again soon?

Reply:
> Do you plan to change this code again soon?
No. Sorry again for  not planning this better.

Reply:
No problem. In fact, I think it is good that your changes get in first. It would have influenced the best configuration that I'm computing.

I did a comment on that commit. I'm not sure you will receive an email about it. So here it is to be sure you see it: https://github.com/tensorflow/tensorflow/commit/0c3feb1cadf79420b18cc0ce7731476ba45d00df#r37354842
How much speed did you saw with your change?